src/server/pfs/server/driver_file.go

```
@@ -283,7 +290,19 @@ func (d *driver) listFile(ctx context.Context, file *pfs.File, paginationMarker
 		number = math.MaxInt64
 	}
 	if reverse {
-		return d.listFileReverse(ctx, name, s, number, cb)
+		fis := newCircularList(number)
+		if err := s.Iterate(ctx, func(fi *pfs.FileInfo, _ fileset.File) error {
+			if pathIsChild(name, pfsfile.CleanPath(fi.File.Path)) {
+				fis.add(fi)
+			}
+			return nil
+		}); err != nil {
+			return errors.EnsureStack(err)
+		}
+		if err := fis.iterateReverse(cb); err != nil {
```

nit, we could just return the error from the call.